beetsplug/convert.py

```diff
@@ -183,11 +183,11 @@ def encode(self, command, source, dest, pretend=False):
             self._log.info(u'Encoding {0}', util.displayable_path(source))
 
         # Substitute $source and $dest in the argument list.
-        args = shlex.split(command)
+        args = shlex.split(command.decode(ui._arg_encoding()))
         for i, arg in enumerate(args):
             args[i] = Template(arg).safe_substitute({
-                'source': source,
-                'dest': dest,
+                'source': source.decode(ui._arg_encoding()),
+                'dest': dest.decode(ui._arg_encoding()),
             })
```

Comment:
`shlex.split` doesn't work with bytes?! That's truly a crime! :angry:

Perhaps we need to rewrite the templating pieces with `.replace('$source', source)`. Otherwise, I can't see a way to exactly preserve the bytes in the filename when passing them to the command.